server/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -210,28 +218,34 @@
         STARTUP_HEADER,
         STARTUP_BODY,
         MSG_HEADER,
-        MSG_BODY
+        MSG_BODY,
+        CANCEL_BODY
     }
 
     private State state = PRE_STARTUP;
 
     PostgresWireProtocol(SQLOperations sqlOperations,
                          Function<SessionContext, AccessControl> getAcessControl,
                          Authentication authService,
-                         @Nullable SslContextProvider sslContextProvider) {
+                         TransportKillJobsNodeAction transportKillJobsNodeAction,
+                         @Nullable SslContextProvider sslContextProvider,
+                         Map<KeyData, Session> activeSessions) {
```

Comment:
I wonder if we should encapsulate both the map and the transport into a `PgSessions` class that exposes a `add`, `remove` and `cancel`?